### PR TITLE
ARGO-377 Implement Topic Delete Request

### PR DESCRIPTION
**Pull request by kaggis**
## Goal 

Implement topic delete Request. 

Request must adhere to the following signature:
` DELETE /v1/projects/PROJET101/topics/TOPIC101`

- if deleleted return empty response and `CODE 200`
- if not deleted return ErrorMsg and appropriate `CODE`

## Implementations
- [X] Implement RemoveResource in mongo store
- [X] Implement RemoveTopic, Remove sub in mongo store, mock store
- [X] Implement RemoveTopic method in Topics 
- [X] Implement TopicDelete handler
- [X] add unit tests
- [X] update swagger definition

### CODE REVIEW

**Comment by kaggis** [comment]
please review
/cc @pkoro @themiszamani

File: handlers_test.go [hunk 1]
```diff
@@ -106,6 +106,60 @@ func (suite *HandlerTestSuite) TestSubListAll() {
 
 }
 
+func (suite *HandlerTestSuite) TestTopicDelete() {
+
+	req, err := http.NewRequest("DEL", "http://localhost:8080/v1/projects/ARGO/topics/topic1", nil)
```
Comment (pkoro) [inline handlers_test.go]: Change `DEL` to `DELETE` in this line.

File: handlers_test.go [hunk 2]
```diff
@@ -106,6 +106,60 @@ func (suite *HandlerTestSuite) TestSubListAll() {
...
+	w := httptest.NewRecorder()
+	router.HandleFunc("/v1/projects/{project}/topics/{topic}", WrapConfig(TopicDelete, cfgKafka, &brk, str))
+	router.ServeHTTP(w, req)
+	suite.Equal(200, w.Code)
+	suite.Equal(expResp, w.Body.String())
+}
+
+func (suite *HandlerTestSuite) TestTopicDeleteNotfound() {
+
+	req, err := http.NewRequest("DEL", "http://localhost:8080/v1/projects/ARGO/topics/topicFoo", nil)
```
Comment (pkoro) [inline handlers_test.go]: Same comment here.

**Comment by pkoro** [comment]
+1

**Pull request merged**